src/transformers/models/dpt/image_processing_dpt.py

```diff
@@ -426,6 +458,51 @@ def preprocess(
         ]
 
         data = {"pixel_values": images}
+
+        if segmentation_maps is not None:
+            segmentation_maps = [to_numpy_array(segmentation_map) for segmentation_map in segmentation_maps]
+
+            # Add channel dimension if missing - needed for certain transformations
+            if segmentation_maps[0].ndim == 2:
+                added_channel_dim = True
+                segmentation_maps = [segmentation_map[None, ...] for segmentation_map in segmentation_maps]
+                input_data_format = ChannelDimension.FIRST
+            else:
+                added_channel_dim = False
+                if input_data_format is None:
+                    input_data_format = infer_channel_dimension_format(segmentation_maps[0], num_channels=1)
+
+            if do_reduce_labels:
+                segmentation_maps = [self.reduce_label(segmentation_map) for segmentation_map in segmentation_maps]
+
+            if do_resize:
+                segmentation_maps = [
+                    self.resize(
+                        image=segmentation_map,
+                        size=size,
+                        resample=resample,
+                        keep_aspect_ratio=keep_aspect_ratio,
+                        ensure_multiple_of=ensure_multiple_of,
+                        input_data_format=input_data_format,
+                    )
+                    for segmentation_map in segmentation_maps
+                ]
+
+            if do_pad:
+                segmentation_maps = [
+                    self.pad_image(
+                        image=segmentation_map, size_divisor=size_divisor, input_data_format=input_data_format
+                    )
+                    for segmentation_map in segmentation_maps
+                ]
+
+            # Remove extra channel dimension if added for processing
+            if added_channel_dim:
+                segmentation_maps = [segmentation_map.squeeze(0) for segmentation_map in segmentation_maps]
+            segmentation_maps = [segmentation_map.astype(np.int64) for segmentation_map in segmentation_maps]
+
+            data["labels"] = segmentation_maps
```

Comment:
Perfect - if there isn't any difference with Beit, can this be wrapped in a `_preprocess_segmentation_map()` method in a loop, that can be flagged as `# Copied from ...` the beit image processor?

Reply:
Wrapped segmentation map preprocessing code to `_preprocess_segmentation_map()`, and also moved image preprocessing to separate `_preprocess_image()` function and general preprocessing functionality to `_preprocess()` function.